diff -upk.orig apt-0.5.15cnc6.orig/tools/genpkglist.cc apt-0.5.15cnc6/tools/genpkglist.cc
--- apt-0.5.15cnc6.orig/tools/genpkglist.cc	2003-12-23 18:30:14 +0000
+++ apt-0.5.15cnc6/tools/genpkglist.cc	2005-11-25 21:57:03 +0000
@@ -13,6 +13,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <assert.h>
+#include <locale.h>
 
 #include <map>
 #include <iostream>
@@ -464,7 +465,7 @@ int main(int argc, char ** argv) 
    const char *pkgListSuffix = NULL;
    bool pkgListAppend = false;
    
-   putenv("LC_ALL="); // Is this necessary yet (after i18n was supported)?
+   setlocale(LC_ALL, "C");
    for (i = 1; i < argc; i++) {
       if (strcmp(argv[i], "--index") == 0) {
 	 i++;
diff -upk.orig apt-0.5.15cnc6.orig/tools/gensrclist.cc apt-0.5.15cnc6/tools/gensrclist.cc
--- apt-0.5.15cnc6.orig/tools/gensrclist.cc	2003-12-23 18:30:14 +0000
+++ apt-0.5.15cnc6/tools/gensrclist.cc	2005-11-25 21:56:58 +0000
@@ -13,6 +13,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <assert.h>
+#include <locale.h>
 
 #include <map>
 #include <list>
@@ -154,7 +155,7 @@ int main(int argc, char ** argv) 
    const char *srcListSuffix = NULL;
    bool srcListAppend = false;
 
-   putenv("LC_ALL="); // Is this necessary yet (after i18n was supported)?
+   setlocale(LC_ALL, "C");
    for (i = 1; i < argc; i++) {
       if (strcmp(argv[i], "--mapi") == 0) {
 	 mapi = true;
