--- apt-0.3.19cnc53/apt-pkg/rpm/rpmpm.cc~	Tue Nov 13 20:32:08 2001
+++ apt-0.3.19cnc53/apt-pkg/rpm/rpmpm.cc	Fri Nov 16 19:10:19 2001
@@ -402,7 +402,7 @@
    if (WIFEXITED(Status) == 0 || WEXITSTATUS(Status) != 0)
    {
       if (WIFSIGNALED(Status) != 0)
-	  return _error->Error(_("Sub-process %s terminated by signal (%i)") ,Args[0], WTERMSIG(Status) );
+	  return _error->Error(_("Sub-process %s terminated by signal (%s)") ,Args[0], strsignal(WTERMSIG(Status)) );
       
       if (WIFEXITED(Status) != 0)
 	  return _error->Error(_("Sub-process %s returned an error code (%u)"),Args[0],
