diff -ur apt-0.5.4cnc3.orig/tools/genbasedir apt-0.5.4cnc3/tools/genbasedir
--- apt-0.5.4cnc3.orig/tools/genbasedir	Sat Jul 27 13:02:53 2002
+++ apt-0.5.4cnc3/tools/genbasedir	Sat Jul 27 14:05:42 2002
@@ -26,6 +26,8 @@
    genbasedir /home/ftp/pub/conectiva\n\
    genbasedir /home/ftp/pub/conectiva main extra devel\n\
 "
+GENPKGLIST=genpkglist-0.5
+GENSRCLIST=gensrclist-0.5
 
 getsize()
 {
@@ -178,9 +180,9 @@
     fi
 
     if test x$updateinfo = x; then
-    	(cd $basedir; genpkglist $progress $bloat --index /tmp/srcidx.$comp.$$ $topdir $comp)
+    	(cd $basedir; "$GENPKGLIST" $progress $bloat --index /tmp/srcidx.$comp.$$ $topdir $comp)
     else
-    	(cd $basedir; genpkglist $progress $bloat --index /tmp/srcidx.$comp.$$ --info $updateinfo $topdir $comp)
+    	(cd $basedir; "$GENPKGLIST" $progress $bloat --index /tmp/srcidx.$comp.$$ --info $updateinfo $topdir $comp)
     fi
 
     if [ -f $basedir/pkglist.$comp ]; then
@@ -230,9 +232,9 @@
 		srctopdir=`cd $topdir; pwd`
 	fi
     if [ $mapi -ne 0 ]; then
-        (cd $basedir; gensrclist $progress $flat --mapi $srctopdir $comp /tmp/srcidx.$comp.$$)
+        (cd $basedir; "$GENSRCLIST" $progress $flat --mapi $srctopdir $comp /tmp/srcidx.$comp.$$)
     else
-        (cd $basedir; gensrclist $progress $flat $srctopdir $comp /tmp/srcidx.$$)
+        (cd $basedir; "$GENSRCLIST" $progress $flat $srctopdir $comp /tmp/srcidx.$$)
     fi
 
     if [ -f $basedir/srclist.$comp ]; then
