--- apt-0.5.4cnc3.orig/apt-pkg/acquire-item.cc	Sat Jul 27 13:02:53 2002
+++ apt-0.5.4cnc3/apt-pkg/acquire-item.cc	Sat Jul 27 13:34:34 2002
@@ -786,7 +786,10 @@
    {
       if (Md5Hash != MD5)
       {
-	 Status = StatError;
+	if (_config->FindB("Debug::pkgAcquire::Auth", false)) {
+	    cerr << "md5 mismatch: " << Md5Hash << "!=" << MD5 << endl;
+	}
+ 	 Status = StatError;
 	 ErrorText = _("MD5Sum mismatch");
 	 Rename(DestFile,DestFile + ".FAILED");
 	 return;
@@ -916,6 +919,9 @@
    {
       if (Md5Hash != MD5)
       {
+	if (_config->FindB("Debug::pkgAcquire::Auth", false)) {
+	    cerr << "md5 mismatch: " << Md5Hash << "!=" << MD5 << endl;
+	}
 	 Status = StatError;
 	 ErrorText = "MD5Sum mismatch";
 	 Rename(DestFile,DestFile + ".FAILED");
--- apt-0.5.4cnc3.orig/cmdline/apt-get.cc	Sat Jul 27 13:02:53 2002
+++ apt-0.5.4cnc3/cmdline/apt-get.cc	Sat Jul 27 13:41:35 2002
@@ -1709,6 +1709,9 @@
 	     I->Type != "tar")
 	    continue;
 	 
+	if (_config->FindB("Debug::pkgAcquire::Auth",false) == true)
+	    cerr << "I->Path = " << I->Path << ", I->MD5Hash = " << I->MD5Hash << endl;
+
 	 new pkgAcqFile(&Fetcher,Last->Index().ArchiveURI(I->Path),
 			I->MD5Hash,I->Size,
 			Last->Index().SourceInfo(*Last,*I),Src);
