diff -uprk.orig apt-0.5.15cnc5.orig/apt-pkg/rpm/rpmpackagedata.cc apt-0.5.15cnc5/apt-pkg/rpm/rpmpackagedata.cc
--- apt-0.5.15cnc5.orig/apt-pkg/rpm/rpmpackagedata.cc	2003-12-23 23:19:36 +0300
+++ apt-0.5.15cnc5/apt-pkg/rpm/rpmpackagedata.cc	2004-01-16 15:14:46 +0300
@@ -21,7 +21,7 @@ RPMPackageData::RPMPackageData()
 #endif
 {
    // Populate priorities
-   string FileName = _config->FindFile("Dir::Etc::rpmpriorities");
+   string FileName = _config->FindFile("Dir::Etc") + "pkgpriorities";
    FileFd F(FileName, FileFd::ReadOnly);
    if (_error->PendingError()) 
    {
diff -uprk.orig apt-0.5.15cnc5.orig/apt-pkg/rpm/rpmpm.cc apt-0.5.15cnc5/apt-pkg/rpm/rpmpm.cc
--- apt-0.5.15cnc5.orig/apt-pkg/rpm/rpmpm.cc	2003-12-23 23:19:20 +0300
+++ apt-0.5.15cnc5/apt-pkg/rpm/rpmpm.cc	2004-01-16 15:14:33 +0300
@@ -493,7 +493,7 @@ bool pkgRPMExtPM::ExecRPM(Item::RPMOps o
       }	 
    }
 
-   if (_config->FindB("RPM::Order",false) == false)
+   if (_config->FindB("RPM::Order",true) == false)
       Args[n++] = "--noorder";
     
    bool FilesInArgs = true;
diff -uprk.orig apt-0.5.15cnc5.orig/buildlib/archtable apt-0.5.15cnc5/buildlib/archtable
--- apt-0.5.15cnc5.orig/buildlib/archtable	2003-12-23 23:15:23 +0300
+++ apt-0.5.15cnc5/buildlib/archtable	2004-01-16 15:14:33 +0300
@@ -5,8 +5,10 @@
 
 # The left side is a regex for awk
 
-i.86	i386
-pentium	i386
+i[34]86        i386
+i.86   i586
+pentium.*      i586
+athlon i586
 sparc	sparc
 sparc64	sparc
 alpha.*	alpha
