--- apt-0.5.15cnc5/configure.in.orig	2004-02-27 20:36:42 +0300
+++ apt-0.5.15cnc5/configure.in	2004-02-27 20:39:42 +0300
@@ -221,7 +221,7 @@
 AM_CONDITIONAL(COMPILE_MANPAGES, test -n "$DOCBOOK2MAN" -a "$enable_docs" != "no")
 
 AM_GNU_GETTEXT_VERSION([0.11.4])
-AM_GNU_GETTEXT
+AM_GNU_GETTEXT([external])
 
 dnl ah_NUM_PROCS
 rc_GLIBC_VER
--- apt-0.5.15cnc5/Makefile.am.orig	2003-12-23 23:30:18 +0300
+++ apt-0.5.15cnc5/Makefile.am	2004-02-27 20:46:05 +0300
@@ -1,5 +1,5 @@
 
-SUBDIRS = buildlib intl lua apt-pkg methods cmdline tools doc m4 po
+SUBDIRS = buildlib lua apt-pkg methods cmdline tools doc m4 po
 
 ACLOCAL_AMFLAGS = -I m4 -I buildlib
 
