--- apt-0.5.5cnc4.1/configure.in.orig	2003-03-08 03:07:11 +0300
+++ apt-0.5.5cnc4.1/configure.in	2003-03-11 13:08:10 +0300
@@ -125,10 +125,10 @@
 [AC_MSG_RESULT(no)])
 
 dnl Check for apt-shell dependencies
-AC_CHECK_LIB(ncurses,tputs,
+AC_CHECK_LIB(tinfo,tputs,
   [AC_CHECK_HEADER(readline/readline.h,
      [AC_CHECK_LIB(readline,readline,
-        [COMPILE_APTSHELL=yes],,[-lncurses])])])
+        [COMPILE_APTSHELL=yes],,[-ltinfo])])])
 AC_SUBST(COMPILE_APTSHELL)
 
 dnl Converts the ARCH to be something singular for this general CPU family
--- apt-0.5.5cnc4.1/cmdline/makefile.orig	2003-03-08 02:10:32 +0300
+++ apt-0.5.5cnc4.1/cmdline/makefile	2003-03-11 13:08:19 +0300
@@ -27,7 +27,7 @@
 ifdef COMPILE_APTSHELL
 # The apt-shell program
 PROGRAM=apt-shell
-SLIBS = -lapt-pkg -lreadline -lncurses $(RPMLIBS)
+SLIBS = -lapt-pkg -lreadline -ltinfo $(RPMLIBS)
 LIB_MAKES = apt-pkg/makefile
 SOURCE = apt-shell.cc acqprogress.cc
 include $(PROGRAM_H)
