commit 542d44fd82ddc53b779ce4a223f10215f129b7bc
Author: pmatilai <pmatilai>
Date:   Mon Feb 6 23:21:02 2006 +0000

    - enable self-referencing provides and see what breaks...

diff --git a/apt-pkg/pkgcachegen.cc b/apt-pkg/pkgcachegen.cc
index 126f0b2..0e06e40 100644
--- a/apt-pkg/pkgcachegen.cc
+++ b/apt-pkg/pkgcachegen.cc
@@ -481,9 +481,12 @@ bool pkgCacheGenerator::ListParser::NewProvides(pkgCache::VerIterator Ver,
 {
    pkgCache &Cache = Owner->Cache;
 
+// PM:2006-02-07 allow self-referencing provides for now at least...
+#if 0
    // We do not add self referencing provides
    if (Ver.ParentPkg().Name() == PackageName)
       return true;
+#endif
    
    // Get a structure
    unsigned long Provides = Owner->Map.Allocate(sizeof(pkgCache::Provides));
