diff -k.orig -urN apt-0.3.19cnc32/cmdline/apt-cdrom.cc.orig apt-0.3.19cnc32/cmdline/apt-cdrom.cc
--- apt-0.3.19cnc32/cmdline/apt-cdrom.cc.orig	Thu Jan  4 23:26:14 2001
+++ apt-0.3.19cnc32/cmdline/apt-cdrom.cc	Mon Jan 22 17:22:32 2001
@@ -723,7 +723,7 @@
       if (0) 
 	  return _error->Error(_("Unable to locate any package files, perhaps this is not a Debian Disc"));
       else
-	  return _error->Error(_("Unable to locate any package files, perhaps this is not a Conectiva Disc"));
+	  return _error->Error(_("Unable to locate any package files, perhaps this is not a ALT Linux Disc"));
    }
    // Check if the CD is in the database
    string Name;
@@ -748,7 +748,7 @@
       if (_config->FindB("APT::CDROM::Rename",false) == true ||
 	  Name.empty() == true)
       {
-	 cout << _("Please provide a name for this Disc, such as 'MyDistro 6.0 Disk 1'");
+	 cout << _("Please provide a name for this Disc, such as 'Spring 2001 Disk 1'");
 
 	 while (1)
 	 {
