diff -ur apt-0.3.19cnc55~/configure.in apt-0.3.19cnc55/configure.in
--- apt-0.3.19cnc55~/configure.in	Wed Mar  6 20:17:10 2002
+++ apt-0.3.19cnc55/configure.in	Tue Mar 26 21:14:21 2002
@@ -64,20 +64,12 @@
 dnl   AC_MSG_ERROR(failed: I need posix threads, pthread)
 dnl fi
 
-dnl Check for DB2
-AC_CHECK_HEADER(db2/db.h,
-     [AC_CHECK_LIB(db2,db_open,
-        [AC_DEFINE(HAVE_DB2) DB2LIB="-ldb2"])])
-AC_SUBST(DB2LIB)
-
-
 dnl Check for rpm version --akk
 rpm_version="none"
 SAVE_LIBS="$LIBS"
 SAVE_CPPFLAGS="$CPPFLAGS"
 
 CPPFLAGS="$SAVE_CPPFLAGS -I/usr/include/rpm"
-AC_CHECK_HEADERS(db1/db.h)
 AC_CHECK_HEADER(rpm/rpmlib.h, rpm_header_ok=1, rpm_header_ok=0)
 
 if test $rpm_header_ok = 1; then
@@ -85,7 +77,7 @@
       LIBS="$SAVE_LIBS -lrpm -lrpmio -lz -lbz2 -lpopt"
       AC_CHECK_LIB(rpmdb,rpmdbOpen,
         [RPMDBLIBS="-lrpmdb"],
-        [RPMDBLIBS="-ldb-3.1"])
+        [RPMDBLIBS="-ldb"])
 
       LIBS="$SAVE_LIBS $RPMDBLIBS -lrpmio -lz -lbz2 -lpopt"
       AC_CHECK_LIB(rpm,rpmdbGetIteratorOffset,
@@ -100,6 +92,7 @@
               [AC_DEFINE_UNQUOTED(HAVE_RPM, 1) 
 	       RPMLIBS="-lrpm -ldb1 -lz -lbz2 -lpopt"
                SAVE_CPPFLAGS="$SAVE_CPPFLAGS -I/usr/include/rpm"
+	       AC_CHECK_HEADERS(db1/db.h)
 	       rpm_version="3"])
       fi
 fi
diff -ur apt-0.3.19cnc55~/tools/cached_md5.cc apt-0.3.19cnc55/tools/cached_md5.cc
--- apt-0.3.19cnc55~/tools/cached_md5.cc	Sun Feb 17 01:46:11 2002
+++ apt-0.3.19cnc55/tools/cached_md5.cc	Sun Feb 17 01:46:25 2002
@@ -39,6 +39,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <rpm/rpmlib.h>
+#include <rpm/misc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
diff -ur apt-0.3.19cnc55~/tools/hdlist2pkglist.cc apt-0.3.19cnc55/tools/hdlist2pkglist.cc
--- apt-0.3.19cnc55~/tools/hdlist2pkglist.cc	Sun Feb 17 01:46:11 2002
+++ apt-0.3.19cnc55/tools/hdlist2pkglist.cc	Sun Feb 17 01:46:46 2002
@@ -37,6 +37,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <rpmlib.h>
+#include <rpm/misc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
