diff -uprk.orig apt-0.5.5cnc2.orig/configure.in apt-0.5.5cnc2/configure.in
--- apt-0.5.5cnc2.orig/configure.in	2003-02-21 17:06:40 +0300
+++ apt-0.5.5cnc2/configure.in	2003-02-25 16:04:31 +0300
@@ -56,10 +56,12 @@ dnl if test "$PTHREADLIB" != "-lpthread"
 dnl   AC_MSG_ERROR(failed: I need posix threads, pthread)
 dnl fi
 
+DB2LIB=
+dnl ALT: uncomment this to build apt-ftparchive
 dnl Check for DB2
-AC_CHECK_HEADER(db2/db.h,
-     [AC_CHECK_LIB(db2,db_open,
-        [AC_DEFINE(HAVE_DB2) DB2LIB="-ldb2"])])
+dnl AC_CHECK_HEADER(db2/db.h,
+dnl      [AC_CHECK_LIB(db2,db_open,
+dnl         [AC_DEFINE(HAVE_DB2) DB2LIB="-ldb2"])])
 AC_SUBST(DB2LIB)
 
 dnl CNC:2002-07-03
@@ -69,7 +71,6 @@ SAVE_LIBS="$LIBS"
 SAVE_CPPFLAGS="$CPPFLAGS"
 
 CPPFLAGS="$SAVE_CPPFLAGS -I/usr/include/rpm"
-AC_CHECK_HEADERS(db1/db.h)
 AC_CHECK_HEADER(rpm/rpmlib.h, rpm_header_ok=1, rpm_header_ok=0)
 
 if test $rpm_header_ok = 1; then
@@ -77,7 +78,7 @@ if test $rpm_header_ok = 1; then
       LIBS="$SAVE_LIBS -lrpm -lrpmio -lz -lbz2 -lpopt"
       AC_CHECK_LIB(rpmdb,rpmdbOpen,
         [RPMDBLIBS="-lrpmdb"],
-        [RPMDBLIBS="-ldb-3.1"])
+        [RPMDBLIBS="-ldb"])
 
       LIBS="$SAVE_LIBS $RPMDBLIBS -lrpmio -lz -lbz2 -lpopt"
       AC_CHECK_LIB(rpm,rpmdbGetIteratorOffset,
@@ -96,6 +97,7 @@ if test $rpm_header_ok = 1; then
               [AC_DEFINE_UNQUOTED(HAVE_RPM, 1) 
 	       RPMLIBS="-lrpm -ldb1 -lz -lbz2 -lpopt"
                SAVE_CPPFLAGS="$SAVE_CPPFLAGS -I/usr/include/rpm"
+	       AC_CHECK_HEADERS(db1/db.h)
 	       rpm_version="3"])
       fi
 fi
diff -uprk.orig apt-0.5.5cnc2.orig/tools/cached_md5.cc apt-0.5.5cnc2/tools/cached_md5.cc
--- apt-0.5.5cnc2.orig/tools/cached_md5.cc	2003-02-21 17:12:36 +0300
+++ apt-0.5.5cnc2/tools/cached_md5.cc	2003-02-25 16:03:17 +0300
@@ -7,6 +7,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <rpm/rpmlib.h>
+#include <rpm/misc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
diff -uprk.orig apt-0.5.5cnc2.orig/tools/hdlist2pkglist.cc apt-0.5.5cnc2/tools/hdlist2pkglist.cc
--- apt-0.5.5cnc2.orig/tools/hdlist2pkglist.cc	2003-02-21 17:12:30 +0300
+++ apt-0.5.5cnc2/tools/hdlist2pkglist.cc	2003-02-25 16:03:17 +0300
@@ -7,6 +7,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <rpmlib.h>
+#include <rpm/misc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
