--- apt-0.5.4cnc9/apt-pkg/rpm/rpmlistparser.cc.orig	2002-08-05 18:55:56 +0400
+++ apt-0.5.4cnc9/apt-pkg/rpm/rpmlistparser.cc	2002-12-23 17:21:35 +0300
@@ -123,8 +123,6 @@ string rpmListParser::Package()
    bool DupOk = false;
    string Name = str;
    
-   if (strncmp(str,"kernel", 6)==0)
-       DupOk=true;
    for (vector<regex_t*>::iterator I = AllowedDupPackages.begin();
 	I != AllowedDupPackages.end(); I++)
    {
