diff -ur apt-0.3.19cnc55~/apt-pkg/init.cc apt-0.3.19cnc55/apt-pkg/init.cc
--- apt-0.3.19cnc55~/apt-pkg/init.cc	Thu Aug  2 01:35:12 2001
+++ apt-0.3.19cnc55/apt-pkg/init.cc	Thu Mar 21 18:47:24 2002
@@ -38,12 +38,9 @@
       Cnf.Set("Dir::State::status","/var/lib/dpkg/status");
    } else {
       Cnf.Set("Acquire::cdrom::mount", "/mnt/cdrom");
-      Cnf.Set("RPM::AllowedDupPkgs::","^kernel$");
-      Cnf.Set("RPM::AllowedDupPkgs::", "kernel-smp");
-      Cnf.Set("RPM::AllowedDupPkgs::", "kernel-enterprise");
+      Cnf.Set("RPM::AllowedDupPkgs::", "^(NVIDIA_)?(kernel|alsa)[0-9]*($|-up|-smp|-secure|-custom|-enterprise|-BOOT|-tape|-aureal)");
 
-      Cnf.Set("RPM::HoldPkgs::", "kernel-source");
-      Cnf.Set("RPM::HoldPkgs::", "kernel-headers");
+      Cnf.Set("RPM::HoldPkgs::", "^(kernel|alsa)[0-9]*-source");
 
       Cnf.Set("Dir::State::status","/var/lib/rpm/status");
    }
