diff -upk.orig apt-0.5.15lorg2.orig/cmdline/Makefile.am apt-0.5.15lorg2/cmdline/Makefile.am
--- apt-0.5.15lorg2.orig/cmdline/Makefile.am	2006-01-16 23:10:47 +0000
+++ apt-0.5.15lorg2/cmdline/Makefile.am	2006-04-01 15:03:29 +0000
@@ -15,7 +15,7 @@ LDADD = ../apt-pkg/libapt-pkg.la $(RPMLI
 apt_get_SOURCES = apt-get.cc acqprogress.cc acqprogress.h
 apt_cache_SOURCES = apt-cache.cc
 apt_shell_SOURCES = apt-shell.cc acqprogress.cc acqprogress.h
-apt_shell_LDADD = $(LDADD) -lreadline -lncurses
+apt_shell_LDADD = $(LDADD) -lreadline
 apt_config_SOURCES = apt-config.cc
 apt_cdrom_SOURCES = apt-cdrom.cc rpmindexcopy.cc rpmindexcopy.h
 
diff -upk.orig apt-0.5.15lorg2.orig/cmdline/Makefile.in apt-0.5.15lorg2/cmdline/Makefile.in
--- apt-0.5.15lorg2.orig/cmdline/Makefile.in	2006-01-17 00:01:33 +0000
+++ apt-0.5.15lorg2/cmdline/Makefile.in	2006-04-01 15:03:29 +0000
@@ -289,7 +289,7 @@ LDADD = ../apt-pkg/libapt-pkg.la $(RPMLI
 apt_get_SOURCES = apt-get.cc acqprogress.cc acqprogress.h
 apt_cache_SOURCES = apt-cache.cc
 apt_shell_SOURCES = apt-shell.cc acqprogress.cc acqprogress.h
-apt_shell_LDADD = $(LDADD) -lreadline -lncurses
+apt_shell_LDADD = $(LDADD) -lreadline
 apt_config_SOURCES = apt-config.cc
 apt_cdrom_SOURCES = apt-cdrom.cc rpmindexcopy.cc rpmindexcopy.h
 apt_get_static_SOURCES = $(apt_get_SOURCES)
diff -upk.orig apt-0.5.15lorg2.orig/configure.in apt-0.5.15lorg2/configure.in
--- apt-0.5.15lorg2.orig/configure.in	2006-02-06 19:12:19 +0000
+++ apt-0.5.15lorg2/configure.in	2006-04-01 15:05:18 +0000
@@ -134,10 +134,9 @@ else
 fi
 
 dnl Check for apt-shell dependencies
-AC_CHECK_LIB(ncurses,tputs,
-  [AC_CHECK_HEADER(readline/readline.h,
-     [AC_CHECK_LIB(readline,rl_completion_matches,
-        [compile_aptshell=yes],,[-lncurses])])])
+AC_CHECK_HEADER(readline/readline.h,
+   [AC_CHECK_LIB(readline,rl_completion_matches,
+      [compile_aptshell=yes])])
 AM_CONDITIONAL(COMPILE_APTSHELL, test "$compile_aptshell" = "yes")
 
 dnl Converts the ARCH to be something singular for this general CPU family
