diff -uprN apt-0.5.15cnc6.orig/apt-pkg/pkgsystem.h apt-0.5.15cnc6/apt-pkg/pkgsystem.h
--- apt-0.5.15cnc6.orig/apt-pkg/pkgsystem.h	2003-11-25 04:55:37 +1200
+++ apt-0.5.15cnc6/apt-pkg/pkgsystem.h	2004-04-13 19:35:39 +1300
@@ -90,7 +90,7 @@ class pkgSystem
    // Return a list of system index files..
    virtual bool AddStatusFiles(std::vector<pkgIndexFile *> &List) = 0;   
    // CNC:2003-11-21
-   virtual bool AddSourceFiles(std::vector<pkgIndexFile *> &List) {};
+   virtual void AddSourceFiles(std::vector<pkgIndexFile *> &List) {};
    virtual bool FindIndex(pkgCache::PkgFileIterator File,
 			  pkgIndexFile *&Found) const = 0;
    
diff -uprN apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmhandler.cc apt-0.5.15cnc6/apt-pkg/rpm/rpmhandler.cc
--- apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmhandler.cc	2004-03-05 23:29:09 +1200
+++ apt-0.5.15cnc6/apt-pkg/rpm/rpmhandler.cc	2004-04-13 19:36:15 +1300
@@ -87,7 +87,7 @@ bool RPMFileHandler::Jump(unsigned Offse
 {
    if (FD == NULL)
       return false;
-   if (lseek(Fileno(FD),Offset,SEEK_SET) != Offset)
+   if (lseek(Fileno(FD),Offset,SEEK_SET) != (off_t)Offset)
       return false;
    return Skip();
 }
diff -uprN apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmpm.cc apt-0.5.15cnc6/apt-pkg/rpm/rpmpm.cc
--- apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmpm.cc	2004-03-16 10:10:41 +1200
+++ apt-0.5.15cnc6/apt-pkg/rpm/rpmpm.cc	2004-04-13 19:39:29 +1300
@@ -393,7 +393,7 @@ bool pkgRPMExtPM::ExecRPM(Item::RPMOps o
    
    Args[n++] = _config->Find("Dir::Bin::rpm","rpm").c_str();
 
-   bool nodeps;
+   bool nodeps = false;
 
    switch (op)
    {
@@ -416,6 +416,8 @@ bool pkgRPMExtPM::ExecRPM(Item::RPMOps o
 	 operation = "-e";
 	 nodeps = true;
 	 break;
+	default:
+	 return false;
    }
    Args[n++] = operation;
 
@@ -657,8 +659,7 @@ pkgRPMLibPM::~pkgRPMLibPM()
 
 bool pkgRPMLibPM::AddToTransaction(Item::RPMOps op, vector<const char*> &files)
 {
-   int debug = _config->FindB("Debug::pkgRPMPM", false);
-   int n = 0, rc, xx;
+   int rc;
    FD_t fd;
    rpmHeader hdr;
 
@@ -739,7 +740,6 @@ bool pkgRPMLibPM::Process(vector<const c
    int rc = 0;
    bool Success = false;
    bool Interactive = _config->FindB("RPM::Interactive",true);
-   int debug = _config->FindB("Debug::pkgRPMPM", false);
    string Dir = _config->Find("RPM::RootDir");
    rpmReadConfigFiles(NULL, NULL);
 
diff -uprN apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmsystem.cc apt-0.5.15cnc6/apt-pkg/rpm/rpmsystem.cc
--- apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmsystem.cc	2004-03-23 04:51:38 +1200
+++ apt-0.5.15cnc6/apt-pkg/rpm/rpmsystem.cc	2004-04-13 19:40:14 +1300
@@ -251,7 +251,7 @@ bool rpmSystem::AddStatusFiles(vector<pk
 // System::AddSourceFiles - Register aditional source files		/*{{{*/
 // ---------------------------------------------------------------------
 /* */
-bool rpmSystem::AddSourceFiles(vector<pkgIndexFile *> &List)
+void rpmSystem::AddSourceFiles(vector<pkgIndexFile *> &List)
 {
    const Configuration::Item *Top;
    Top = _config->Tree("APT::Arguments");
@@ -268,7 +268,6 @@ bool rpmSystem::AddSourceFiles(vector<pk
 	 }
       }
    }
-   return true;
 }
 									/*}}}*/
 #ifdef OLD_FILEDEPS
diff -uprN apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmsystem.h apt-0.5.15cnc6/apt-pkg/rpm/rpmsystem.h
--- apt-0.5.15cnc6.orig/apt-pkg/rpm/rpmsystem.h	2003-11-25 04:55:37 +1200
+++ apt-0.5.15cnc6/apt-pkg/rpm/rpmsystem.h	2004-04-13 19:40:53 +1300
@@ -60,7 +60,7 @@ class rpmSystem : public pkgSystem
    virtual bool ArchiveSupported(const char *Type);
    virtual signed Score(Configuration const &Cnf);
    virtual bool AddStatusFiles(vector<pkgIndexFile *> &List);
-   virtual bool AddSourceFiles(vector<pkgIndexFile *> &List);
+   virtual void AddSourceFiles(vector<pkgIndexFile *> &List);
    virtual bool FindIndex(pkgCache::PkgFileIterator File,
 			  pkgIndexFile *&Found) const;
    virtual bool ProcessCache(pkgDepCache &Cache,pkgProblemResolver &Fix);
