--- apt-0.5.5cnc1/apt-pkg/deb/debsystem.cc.orig	2002-07-23 21:54:51 +0400
+++ apt-0.5.5cnc1/apt-pkg/deb/debsystem.cc	2003-02-13 16:05:56 +0300
@@ -185,6 +185,8 @@
 signed debSystem::Score(Configuration const &Cnf)
 {
    signed Score = 0;
+   if (Cnf.FindB("APT::Ignore-dpkg",true) == true)
+      return Score;
    if (FileExists(Cnf.FindFile("Dir::State::status","/var/lib/dpkg/status")) == true)
        Score += 10;
    if (FileExists(Cnf.FindFile("Dir::Bin::dpkg","/usr/bin/dpkg")) == true)
