diff -uprk.orig apt-0.5.15cnc5.orig/configure.in apt-0.5.15cnc5/configure.in
--- apt-0.5.15cnc5.orig/configure.in	2003-12-23 23:15:40 +0300
+++ apt-0.5.15cnc5/configure.in	2004-01-15 19:08:18 +0300
@@ -72,7 +72,7 @@ else
 		[AC_MSG_ERROR([Can't find libbz2 library])])
    AC_CHECK_LIB(z,gzopen, [],
 		[AC_MSG_ERROR([Can't find libz library])])
-   AC_CHECK_HEADERS(db1/db.h)
+   dnl AC_CHECK_HEADERS(db1/db.h)
 
    CPPFLAGS="$CPPFLAGS -I/usr/include/rpm"
    AC_CHECK_HEADER(rpm/rpmlib.h, [],
@@ -80,12 +80,13 @@ else
 
    if test $RPM_VERSION_MAJOR -lt 4; then
       RPMLIBS="-lrpm -lz -lbz2 -lpopt"
+      AC_CHECK_HEADERS(db1/db.h)
    else
       AC_CHECK_LIB(rpmdb,rpmdbOpen,
          [RPMDBLIBS="-lrpmdb"],
-         [RPMDBLIBS="-ldb-3.1"],
+         [RPMDBLIBS="-ldb"],
          [-lrpm -lrpmio -lz -lbz2 -lpopt])
-      RPMLIBS="-lrpm $RPMDBLIBS -lrpmio -lz -lbz2 -lpopt"
+      RPMLIBS="-lrpm -lrpmio $RPMDBLIBS"
    fi
    AC_MSG_CHECKING(for RPM libraries)
    AC_MSG_RESULT($RPMLIBS)
diff -uprk.orig apt-0.5.15cnc5.orig/tools/cached_md5.cc apt-0.5.15cnc5/tools/cached_md5.cc
--- apt-0.5.15cnc5.orig/tools/cached_md5.cc	2003-12-23 23:16:37 +0300
+++ apt-0.5.15cnc5/tools/cached_md5.cc	2004-01-15 19:07:44 +0300
@@ -7,6 +7,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <rpm/rpmlib.h>
+#include <rpm/misc.h>
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
