--- apt-0.5.5cnc1/apt-pkg/rpm/rpmsystem.cc.orig	2002-11-25 21:25:28 +0300
+++ apt-0.5.5cnc1/apt-pkg/rpm/rpmsystem.cc	2003-02-07 18:13:09 +0300
@@ -125,7 +125,7 @@
    Cnf.CndSet("Dir::Etc::translateparts", "translate.list.d");
    Cnf.CndSet("Dir::State::prefetch", "prefetch");
    Cnf.CndSet("Dir::Locale","/usr/share/locale");
-   Cnf.CndSet("Acquire::DistroID","Conectiva"); // hee hee
+   Cnf.CndSet("Acquire::DistroID","ALT Linux"); // hee hee
    Cnf.CndSet("Acquire::CDROM::Mount", "/mnt/cdrom");
    Cnf.CndSet("Acquire::CDROM::Copy-All", "true");
 
--- apt-0.5.5cnc2/cmdline/apt-cdrom.cc.orig	2003-02-21 17:09:14 +0300
+++ apt-0.5.5cnc2/cmdline/apt-cdrom.cc	2003-02-25 15:33:02 +0300
@@ -700,7 +700,7 @@ bool DoAdd(CommandLine &)
       {
 // CNC:2002-07-11
 #ifdef HAVE_RPM
-	 cout << "Please provide a name for this Disc, such as 'Conectiva Disk 1'";
+	 cout << "Please provide a name for this Disc, such as 'ALT Linux Disk 1'";
 #else
 	 cout << "Please provide a name for this Disc, such as 'Debian 2.1r1 Disk 1'";
 #endif
--- apt-0.5.5cnc2/methods/ftp.cc.orig	2003-02-21 17:12:17 +0300
+++ apt-0.5.5cnc2/methods/ftp.cc	2003-02-25 16:02:07 +0300
@@ -189,7 +189,7 @@ bool FTPConn::Login()
    
    // Setup the variables needed for authentication
    string User = "anonymous";
-   string Pass = "apt_get_ftp_2.1@debian.linux.user";
+   string Pass = "apt_get_ftp_2.1@alt.linux.user";
 
    // Fill in the user/pass
    if (ServerName.User.empty() == false)
