--- apt-0.5.5cnc1_rc2/buildlib/environment.mak.in.orig	2003-02-02 12:34:15.000000000 -0500
+++ apt-0.5.5cnc1_rc2/buildlib/environment.mak.in	2003-02-02 12:35:17.000000000 -0500
@@ -35,7 +35,7 @@
 # SGML for the man pages
 NSGMLS = @NSGMLS@
 SGMLSPL = @SGMLSPL@
-DOCBOOK2MAN := $(wildcard /usr/lib/perl5/sgmlspl-specs/docbook2man-spec.pl)
+DOCBOOK2MAN := $(wildcard /usr/share/sgml/docbook/utils-*/helpers/docbook2man-spec.pl)
 
 # Gettext settings
 GMSGFMT = @GMSGFMT@
