--- apt-0.3.19cnc55~/apt-pkg/rpm/rpmpm.cc	Wed Mar  6 20:17:13 2002
+++ apt-0.3.19cnc55/apt-pkg/rpm/rpmpm.cc	Thu Mar 21 22:35:05 2002
@@ -287,14 +287,16 @@
     case OInstall:
       options = "-i";
 	    
-      Args[n++] = "-i";
+      Args[n++] = "-iv";
       
       Args[n++] = "--replacepkgs";
 
       if (noninteractive)
 	 Args[n++] = "--percent";
-      else
+      else {
 	 Args[n++] = "-h";
+	 Args[n++] = "--fancypercent";
+      }
       
       if (_config->FindB("RPM::Force", false) == true) 
 	  Args[n++] = "--force";
@@ -309,8 +311,10 @@
 
       if (noninteractive)
 	 Args[n++] = "--percent";
-      else
+      else {
 	 Args[n++] = "-h";
+	 Args[n++] = "--fancypercent";
+      }
       
       if (_config->FindB("RPM::Force", false) == true)
 	  Args[n++] = "--force";
@@ -330,6 +334,13 @@
    if (nodeps)
        Args[n++] = "--nodeps";
 
+    string cmd;
+    for (unsigned i = 0; i < n; ++i)
+    {
+        if (cmd.length())
+            cmd += ' ';
+        cmd += Args[i];
+    }
    
    for (slist<char*>::iterator i = files->begin();
 	i != files->end() && n < sizeof(Args);
@@ -349,7 +360,7 @@
       return true;
    }
 
-   cout << _("Executing RPM (")<<options<<")..." << endl;
+   cout << _("Executing RPM (") << cmd << ")..." << endl;
 
    cout << flush;
    clog << flush;
