
https://bugzilla.altlinux.org/show_bug.cgi?id=7618

--- apt/cmdline/apt-shell.cc.orig	2007-07-31 23:23:46 +0700
+++ apt/cmdline/apt-shell.cc	2007-07-31 23:42:10 +0700
@@ -824,7 +824,7 @@ bool ShowChanges(CacheFile &Cache,pkgDep
 	 ioprintf(c1out,_("After unpacking will need more %sB of disk space.\n"),
 		  SizeToStr(UsrSize).c_str());
       else
-	 ioprintf(c1out,_("After unpacking will need less %sB of disk space.\n"),
+	 ioprintf(c1out,_("After unpacking %sB disk space will be freed.\n"),
 		  SizeToStr(-1*UsrSize).c_str());
    } else {
       double DebBytes = Cache->DebSize();
