--- apt-0.5.15lorg2.orig/apt-pkg/acquire.cc
+++ apt-0.5.15lorg2/apt-pkg/acquire.cc
@@ -16,6 +16,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/acquire.h"
 #endif       
+#include <config.h>
+
 #include <apt-pkg/acquire.h>
 #include <apt-pkg/acquire-item.h>
 #include <apt-pkg/acquire-worker.h>
--- apt-0.5.15lorg2.orig/apt-pkg/acquire-item.cc
+++ apt-0.5.15lorg2/apt-pkg/acquire-item.cc
@@ -16,6 +16,9 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/acquire-item.h"
 #endif
+
+#include <config.h>
+
 #include <apt-pkg/acquire-item.h>
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/sourcelist.h>
@@ -26,7 +29,6 @@
 // CNC:2002-07-03
 #include <apt-pkg/repository.h>
 #include <apt-pkg/md5.h>
-#include <config.h>
 #include <apt-pkg/luaiface.h>
 #include <iostream>
 #include <assert.h>
--- apt-0.5.15lorg2.orig/apt-pkg/acquire-method.cc
+++ apt-0.5.15lorg2/apt-pkg/acquire-method.cc
@@ -18,6 +18,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/acquire-method.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/configuration.h>
--- apt-0.5.15lorg2.orig/apt-pkg/acquire-worker.cc
+++ apt-0.5.15lorg2/apt-pkg/acquire-worker.cc
@@ -15,6 +15,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/acquire-worker.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/acquire-worker.h>
 #include <apt-pkg/acquire-item.h>
 #include <apt-pkg/configuration.h>
--- apt-0.5.15lorg2.orig/apt-pkg/algorithms.cc
+++ apt-0.5.15lorg2/apt-pkg/algorithms.cc
@@ -17,6 +17,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/algorithms.h"
 #endif 
+#include <config.h>
+
 #include <apt-pkg/algorithms.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/configuration.h>
--- apt-0.5.15lorg2.orig/apt-pkg/cachefile.cc
+++ apt-0.5.15lorg2/apt-pkg/cachefile.cc
@@ -16,6 +16,8 @@
 #pragma implementation "apt-pkg/cachefile.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/cachefile.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/sourcelist.h>
--- apt-0.5.15lorg2.orig/apt-pkg/clean.cc
+++ apt-0.5.15lorg2/apt-pkg/clean.cc
@@ -12,6 +12,8 @@
 #pragma implementation "apt-pkg/clean.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/clean.h>
 #include <apt-pkg/strutl.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/cmndline.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/cmndline.cc
@@ -14,6 +14,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/cmndline.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/cmndline.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/strutl.h>
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/configuration.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/configuration.cc
@@ -18,6 +18,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/configuration.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/strutl.h>
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/crc-16.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/crc-16.cc
@@ -19,6 +19,8 @@
 #pragma implementation "apt-pkg/crc-16.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/crc-16.h>
 									/*}}}*/
 
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/error.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/error.cc
@@ -18,6 +18,8 @@
 #pragma implementation "apt-pkg/error.h"
 #endif 
 
+#include "config.h"
+   									/*}}}*/
 #include <apt-pkg/error.h>
 
 #include <iostream>
@@ -28,8 +30,6 @@
 #include <unistd.h>
 #include <cstring>
 
-#include "config.h"
-   									/*}}}*/
 
 using namespace std;
 
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/fileutl.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/fileutl.cc
@@ -17,6 +17,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/fileutl.h"
 #endif 
+#include <config.h>
+
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/sptr.h>
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/hashes.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/hashes.cc
@@ -15,6 +15,8 @@
 #pragma implementation "apt-pkg/hashes.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/hashes.h>
     
 #include <unistd.h>    
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/md5.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/md5.cc
@@ -39,6 +39,8 @@
 #pragma implementation "apt-pkg/md5.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/md5.h>
 #include <apt-pkg/strutl.h>
 
@@ -46,7 +48,6 @@
 #include <unistd.h>
 #include <netinet/in.h>                          // For htonl
 #include <inttypes.h>
-#include <config.h>
 #include <system.h>
 									/*}}}*/
 
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/mmap.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/mmap.cc
@@ -25,6 +25,8 @@
 #pragma implementation "apt-pkg/mmap.h"
 #endif 
 
+#include <config.h>
+
 #define _BSD_SOURCE
 #include <apt-pkg/mmap.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/progress.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/progress.cc
@@ -11,6 +11,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/progress.h"
 #endif 
+#include <config.h>
+
 #include <apt-pkg/progress.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/configuration.h>
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/sha1.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/sha1.cc
@@ -33,13 +33,14 @@
 #pragma implementation "apt-pkg/sha1.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/sha1.h>
 #include <apt-pkg/strutl.h>
 
 #include <string.h>
 #include <unistd.h>
 #include <inttypes.h>
-#include <config.h>
 #include <system.h>
 									/*}}}*/
 
--- apt-0.5.15lorg2.orig/apt-pkg/contrib/strutl.cc
+++ apt-0.5.15lorg2/apt-pkg/contrib/strutl.cc
@@ -19,6 +19,8 @@
 #pragma implementation "apt-pkg/strutl.h"
 #endif
 
+#include "config.h"
+
 #include <apt-pkg/strutl.h>
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/error.h>
@@ -33,8 +35,6 @@
 #include <errno.h>
 #include <stdarg.h>
 
-#include "config.h"
-
 using namespace std;
 									/*}}}*/
 
--- apt-0.5.15lorg2.orig/apt-pkg/depcache.cc
+++ apt-0.5.15lorg2/apt-pkg/depcache.cc
@@ -11,6 +11,9 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/depcache.h"
 #endif
+
+#include <config.h>
+
 #include <apt-pkg/depcache.h>
 #include <apt-pkg/version.h>
 #include <apt-pkg/error.h>
@@ -24,7 +27,6 @@
 #include <apt-pkg/pkgsystem.h>
 
 // CNC:2003-03-17
-#include <config.h>
 #include <apt-pkg/luaiface.h>
     
 #include <apti18n.h>    
--- apt-0.5.15lorg2.orig/apt-pkg/indexfile.cc
+++ apt-0.5.15lorg2/apt-pkg/indexfile.cc
@@ -12,6 +12,8 @@
 #pragma implementation "apt-pkg/indexfile.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/indexfile.h>
 #include <apt-pkg/error.h>
 #include <cstring>
--- apt-0.5.15lorg2.orig/apt-pkg/init.cc
+++ apt-0.5.15lorg2/apt-pkg/init.cc
@@ -8,12 +8,12 @@
    ##################################################################### */
 									/*}}}*/
 // Include files							/*{{{*/
+#include <config.h>
 #include <apt-pkg/init.h>
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/error.h>
 
 #include <apti18n.h>
-#include <config.h>
 #include <sys/stat.h>
 #include <sys/utsname.h>
 
--- apt-0.5.15lorg2.orig/apt-pkg/Makefile.am
+++ apt-0.5.15lorg2/apt-pkg/Makefile.am
@@ -2,7 +2,7 @@
 lib_LTLIBRARIES = libapt-pkg.la
 
 libapt_pkg_la_LIBADD = @RPMLIBS@
-libapt_pkg_la_LDFLAGS = -version-info 3:0:1 -release @GLIBC_VER@-@LIBSTDCPP_VER@
+libapt_pkg_la_LDFLAGS = -version-info 3:0:1 -release @GLIBC_VER@-@LIBSTDCPP_VER@@FILE_OFFSET_BITS_SUFFIX@
 
 AM_CPPFLAGS = -DLIBDIR=\"$(libdir)\"
 
--- apt-0.5.15lorg2.orig/apt-pkg/orderlist.cc
+++ apt-0.5.15lorg2/apt-pkg/orderlist.cc
@@ -66,6 +66,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/orderlist.h"
 #endif 
+#include <config.h>
+
 #include <apt-pkg/orderlist.h>
 #include <apt-pkg/depcache.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/apt-pkg/packagemanager.cc
+++ apt-0.5.15lorg2/apt-pkg/packagemanager.cc
@@ -17,6 +17,8 @@
 #pragma implementation "apt-pkg/packagemanager.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/packagemanager.h>
 #include <apt-pkg/orderlist.h>
 #include <apt-pkg/depcache.h>
--- apt-0.5.15lorg2.orig/apt-pkg/pkgcache.cc
+++ apt-0.5.15lorg2/apt-pkg/pkgcache.cc
@@ -25,6 +25,8 @@
 #pragma implementation "apt-pkg/cacheiterators.h"
 #endif 
 
+#include <config.h>
+
 #define PKGCACHE_FINDPKG_ABI
 #include <apt-pkg/pkgcache.h>
 #include <apt-pkg/version.h>
--- apt-0.5.15lorg2.orig/apt-pkg/pkgcachegen.cc
+++ apt-0.5.15lorg2/apt-pkg/pkgcachegen.cc
@@ -16,6 +16,8 @@
 
 #define APT_COMPATIBILITY 986
 
+#include <config.h>
+
 #include <apt-pkg/pkgcachegen.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/version.h>
--- apt-0.5.15lorg2.orig/apt-pkg/pkgrecords.cc
+++ apt-0.5.15lorg2/apt-pkg/pkgrecords.cc
@@ -12,6 +12,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/pkgrecords.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/pkgrecords.h>
 #include <apt-pkg/indexfile.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/apt-pkg/pkgsystem.cc
+++ apt-0.5.15lorg2/apt-pkg/pkgsystem.cc
@@ -14,6 +14,8 @@
 #pragma implementation "apt-pkg/pkgsystem.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/pkgsystem.h>
 #include <apt-pkg/policy.h>
 #include <cstring>
--- apt-0.5.15lorg2.orig/apt-pkg/policy.cc
+++ apt-0.5.15lorg2/apt-pkg/policy.cc
@@ -26,6 +26,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/policy.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/policy.h>
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/tagfile.h>
--- apt-0.5.15lorg2.orig/apt-pkg/repository.cc
+++ apt-0.5.15lorg2/apt-pkg/repository.cc
@@ -14,6 +14,8 @@
 #pragma implementation "apt-pkg/repository.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/sourcelist.h>
 #include <apt-pkg/repository.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/apt-pkg/rpm/rpmhandler.h
+++ apt-0.5.15lorg2/apt-pkg/rpm/rpmhandler.h
@@ -10,13 +10,15 @@
 #ifndef PKGLIB_RPMHANDLER_H
 #define PKGLIB_RPMHANDLER_H
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
 #include <apt-pkg/fileutl.h>
 
 #include <rpm/rpmlib.h>
 #include <rpm/rpmmacro.h>
 
-#include <config.h>
-
 class RPMHandler
 {
    protected:
--- apt-0.5.15lorg2.orig/apt-pkg/sourcelist.cc
+++ apt-0.5.15lorg2/apt-pkg/sourcelist.cc
@@ -12,6 +12,8 @@
 #pragma implementation "apt-pkg/sourcelist.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/sourcelist.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/fileutl.h>
--- apt-0.5.15lorg2.orig/apt-pkg/srcrecords.cc
+++ apt-0.5.15lorg2/apt-pkg/srcrecords.cc
@@ -15,6 +15,8 @@
 #pragma implementation "apt-pkg/srcrecords.h"
 #endif 
 
+#include <config.h>
+
 #include <apt-pkg/srcrecords.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/sourcelist.h>
--- apt-0.5.15lorg2.orig/apt-pkg/tagfile.cc
+++ apt-0.5.15lorg2/apt-pkg/tagfile.cc
@@ -15,6 +15,8 @@
 #pragma implementation "apt-pkg/tagfile.h"
 #endif
 
+#include <config.h>
+
 #include <apt-pkg/tagfile.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/strutl.h>
--- apt-0.5.15lorg2.orig/apt-pkg/version.cc
+++ apt-0.5.15lorg2/apt-pkg/version.cc
@@ -12,6 +12,8 @@
 #pragma implementation "apt-pkg/version.h"
 #endif 
 
+#include <config.h>
+
 #include <apt-pkg/version.h>
 #include <apt-pkg/pkgcache.h>
 
--- apt-0.5.15lorg2.orig/apt-pkg/versionmatch.cc
+++ apt-0.5.15lorg2/apt-pkg/versionmatch.cc
@@ -14,6 +14,8 @@
 #ifdef __GNUG__
 #pragma implementation "apt-pkg/versionmatch.h"
 #endif
+#include <config.h>
+
 #include <apt-pkg/versionmatch.h>
 // CNC:2003-11-05
 #include <apt-pkg/version.h>
--- apt-0.5.15lorg2.orig/cmdline/acqprogress.cc
+++ apt-0.5.15lorg2/cmdline/acqprogress.cc
@@ -8,6 +8,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include files							/*{{{*/
+#include <config.h>
+
 #include "acqprogress.h"
 #include <apt-pkg/acquire-item.h>
 #include <apt-pkg/acquire-worker.h>
--- apt-0.5.15lorg2.orig/cmdline/apt-cache.cc
+++ apt-0.5.15lorg2/cmdline/apt-cache.cc
@@ -13,6 +13,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/error.h>
 #include <apt-pkg/pkgcachegen.h>
 #include <apt-pkg/init.h>
@@ -28,7 +30,6 @@
 #include <apt-pkg/algorithms.h>
 #include <apt-pkg/sptr.h>
 
-#include <config.h>
 #include <apti18n.h>
 
 // CNC:2003-02-14 - apti18n.h includes libintl.h which includes locale.h,
--- apt-0.5.15lorg2.orig/cmdline/apt-cdrom.cc
+++ apt-0.5.15lorg2/cmdline/apt-cdrom.cc
@@ -11,6 +11,7 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
 #include <apt-pkg/cmndline.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/init.h>
@@ -18,7 +19,6 @@
 #include <apt-pkg/progress.h>
 #include <apt-pkg/cdromutl.h>
 #include <apt-pkg/strutl.h>
-#include <config.h>
 #include <apti18n.h>
     
 // CNC:2002-07-11
--- apt-0.5.15lorg2.orig/cmdline/apt-config.cc
+++ apt-0.5.15lorg2/cmdline/apt-config.cc
@@ -16,12 +16,13 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/cmndline.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/init.h>
 #include <apt-pkg/strutl.h>
 
-#include <config.h>
 #include <apti18n.h>
 
 // CNC:2003-02-14 - apti18n.h includes libintl.h which includes locale.h,
--- apt-0.5.15lorg2.orig/cmdline/apt-get.cc
+++ apt-0.5.15lorg2/cmdline/apt-get.cc
@@ -25,6 +25,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/error.h>
 #include <apt-pkg/cmndline.h>
 #include <apt-pkg/init.h>
@@ -40,7 +42,6 @@
 #include <apt-pkg/sptr.h>
 #include <apt-pkg/versionmatch.h>
     
-#include <config.h>
 #include <apti18n.h>
 
 #include "acqprogress.h"
--- apt-0.5.15lorg2.orig/cmdline/apt-pipe.c
+++ apt-0.5.15lorg2/cmdline/apt-pipe.c
@@ -6,6 +6,8 @@
 #define APT_PIPE_PATH "/var/lib/apt/pipe"
 #endif
 
+#include <config.h>
+
 #include <argz.h>
 #include <errno.h>
 #include <fcntl.h>
--- apt-0.5.15lorg2.orig/cmdline/apt-shell.cc
+++ apt-0.5.15lorg2/cmdline/apt-shell.cc
@@ -25,6 +25,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/error.h>
 #include <apt-pkg/cmndline.h>
 #include <apt-pkg/init.h>
@@ -40,7 +42,6 @@
 #include <apt-pkg/sptr.h>
 #include <apt-pkg/versionmatch.h>
 
-#include <config.h>
 #include <apti18n.h>
 
 // CNC:2003-03-17
--- apt-0.5.15lorg2.orig/cmdline/indexcopy.cc
+++ apt-0.5.15lorg2/cmdline/indexcopy.cc
@@ -10,6 +10,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include "indexcopy.h"
 
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/cmdline/rpmindexcopy.cc
+++ apt-0.5.15lorg2/cmdline/rpmindexcopy.cc
@@ -1,4 +1,6 @@
 
+#include <config.h>
+
 #include <apt-pkg/error.h>
 #include <apt-pkg/progress.h>
 #include <apt-pkg/strutl.h>
--- apt-0.5.15lorg2.orig/configure.in
+++ apt-0.5.15lorg2/configure.in
@@ -19,6 +19,13 @@ AC_ISC_POSIX
 dnl Check for other programs
 AC_PROG_CXX
 AC_PROG_CPP
+AC_SYS_LARGEFILE
+
+case "$ac_cv_sys_file_offset_bits" in
+	no | unknown) FILE_OFFSET_BITS_SUFFIX=;;
+	*) FILE_OFFSET_BITS_SUFFIX="-$ac_cv_sys_file_offset_bits";;
+esac
+AC_SUBST(FILE_OFFSET_BITS_SUFFIX)
 
 AC_PROG_LIBTOOL
 
--- apt-0.5.15lorg2.orig/methods/cdrom.cc
+++ apt-0.5.15lorg2/methods/cdrom.cc
@@ -8,6 +8,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/cdromutl.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/methods/connect.cc
+++ apt-0.5.15lorg2/methods/connect.cc
@@ -11,6 +11,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include "connect.h"
 #include <apt-pkg/error.h>
 #include <apt-pkg/fileutl.h>
--- apt-0.5.15lorg2.orig/methods/copy.cc
+++ apt-0.5.15lorg2/methods/copy.cc
@@ -9,6 +9,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/methods/file.cc
+++ apt-0.5.15lorg2/methods/file.cc
@@ -13,6 +13,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/error.h>
 
--- apt-0.5.15lorg2.orig/methods/ftp.cc
+++ apt-0.5.15lorg2/methods/ftp.cc
@@ -15,6 +15,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/methods/gpg.cc
+++ apt-0.5.15lorg2/methods/gpg.cc
@@ -1,4 +1,6 @@
 
+#include <config.h>
+
 #include <apt-pkg/error.h>
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/strutl.h>
--- apt-0.5.15lorg2.orig/methods/gzip.cc
+++ apt-0.5.15lorg2/methods/gzip.cc
@@ -9,6 +9,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/acquire-method.h>
--- apt-0.5.15lorg2.orig/methods/http.cc
+++ apt-0.5.15lorg2/methods/http.cc
@@ -25,6 +25,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/methods/rfc2553emu.cc
+++ apt-0.5.15lorg2/methods/rfc2553emu.cc
@@ -14,6 +14,8 @@
 
    ##################################################################### */
 									/*}}}*/
+#include <config.h>
+
 #include "rfc2553emu.h"
 #include <stdlib.h>
 #include <arpa/inet.h>
--- apt-0.5.15lorg2.orig/methods/rsh.cc
+++ apt-0.5.15lorg2/methods/rsh.cc
@@ -11,6 +11,8 @@
    ##################################################################### */
 									/*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include "rsh.h"
 #include <apt-pkg/error.h>
 
--- apt-0.5.15lorg2.orig/methods/rsync.cc
+++ apt-0.5.15lorg2/methods/rsync.cc
@@ -8,6 +8,8 @@ RSYNC Aquire Method - This is the RSYNC
 ##################################################################### */
 /*}}}*/
 // Include Files							/*{{{*/
+#include <config.h>
+
 #include <apt-pkg/fileutl.h>
 #include <apt-pkg/acquire-method.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/test/conf.cc
+++ apt-0.5.15lorg2/test/conf.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/error.h>
 
--- apt-0.5.15lorg2.orig/test/extract-control.cc
+++ apt-0.5.15lorg2/test/extract-control.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/debfile.h>
 #include <apt-pkg/error.h>
 
--- apt-0.5.15lorg2.orig/test/hash.cc
+++ apt-0.5.15lorg2/test/hash.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/md5.h>
 #include <apt-pkg/sha1.h>
 #include <apt-pkg/strutl.h>
--- apt-0.5.15lorg2.orig/test/mthdcat.cc
+++ apt-0.5.15lorg2/test/mthdcat.cc
@@ -2,6 +2,8 @@
    All this does is cat a file into the method without closing the FD when
    the file ends */
 
+#include <config.h>
+
 #include <unistd.h>
 
 int main()
--- apt-0.5.15lorg2.orig/test/rpmver.cc
+++ apt-0.5.15lorg2/test/rpmver.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/debversion.h>
 #include <rpm/rpmio.h>
 #include <rpm/misc.h>
--- apt-0.5.15lorg2.orig/test/scratch.cc
+++ apt-0.5.15lorg2/test/scratch.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/dpkgdb.h>
 #include <apt-pkg/debfile.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/test/testdeb.cc
+++ apt-0.5.15lorg2/test/testdeb.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/dirstream.h>
 #include <apt-pkg/debfile.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/test/testextract.cc
+++ apt-0.5.15lorg2/test/testextract.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <apt-pkg/dpkgdb.h>
 #include <apt-pkg/debfile.h>
 #include <apt-pkg/error.h>
--- apt-0.5.15lorg2.orig/test/uri.cc
+++ apt-0.5.15lorg2/test/uri.cc
@@ -1,3 +1,5 @@
+#include <config.h>
+
 #include <apt-pkg/strutl.h>
 #include <stdio.h>
 
--- apt-0.5.15lorg2.orig/test/versiontest.cc
+++ apt-0.5.15lorg2/test/versiontest.cc
@@ -14,6 +14,8 @@
    
    ##################################################################### */
 									/*}}}*/
+#include <config.h>
+
 #include <system.h>
 #include <apt-pkg/error.h>
 #include <apt-pkg/version.h>
--- apt-0.5.15lorg2.orig/tools/cached_md5.cc
+++ apt-0.5.15lorg2/tools/cached_md5.cc
@@ -1,6 +1,8 @@
 /*
  * $Id: cached_md5.cc,v 1.4 2003/01/29 13:47:31 niemeyer Exp $
  */
+#include <config.h>
+
 #include <alloca.h>
 #include <ctype.h>
 #include <dirent.h>
@@ -22,8 +24,6 @@
 #include <apt-pkg/configuration.h>
 #include <apt-pkg/md5.h>
 
-#include <config.h>
-
 CachedMD5::CachedMD5(string DirName, string Domain)
 {
    string fname = DirName;
--- apt-0.5.15lorg2.orig/tools/countpkglist.cc
+++ apt-0.5.15lorg2/tools/countpkglist.cc
@@ -1,3 +1,6 @@
+
+#include <config.h>
+
 #include <stdio.h>
 #include <errno.h>
 #include <string.h>
--- apt-0.5.15lorg2.orig/tools/genpkglist.cc
+++ apt-0.5.15lorg2/tools/genpkglist.cc
@@ -1,6 +1,8 @@
 /*
  * $Id: genpkglist.cc,v 1.7 2003/01/30 17:18:21 niemeyer Exp $
  */
+#include <config.h>
+
 #include <alloca.h>
 #include <ctype.h>
 #include <dirent.h>
--- apt-0.5.15lorg2.orig/tools/gensrclist.cc
+++ apt-0.5.15lorg2/tools/gensrclist.cc
@@ -1,6 +1,8 @@
 /*
  * $Id: gensrclist.cc,v 1.8 2003/01/30 17:18:21 niemeyer Exp $
  */
+#include <config.h>
+
 #include <alloca.h>
 #include <ctype.h>
 #include <dirent.h>
