diff -ur apt-0.5.5cnc4.1/apt-pkg/algorithms.cc apt-0.5.5cnc4.1_20030324a/apt-pkg/algorithms.cc
--- apt-0.5.5cnc4.1/apt-pkg/algorithms.cc	2003-03-08 02:12:08 +0300
+++ apt-0.5.5cnc4.1_20030324a/apt-pkg/algorithms.cc	2003-03-24 20:27:31 +0300
@@ -419,7 +419,8 @@
    if (_config->FindB("APT::Remove-Depends",false) == true)
       Fix.RemoveDepends();
    
-   return Fix.Resolve();
+   // CNC:2003-03-22
+   return Fix.Resolve(true);
 }
 									/*}}}*/
 // AllUpgrade - Upgrade as many packages as possible			/*{{{*/
@@ -1029,6 +1030,22 @@
 		  if ((Flags[Pkg->ID] & Protected) != 0)
 		     continue;
 		
+		  // CNC:2003-03-22
+		  pkgDepCache::State State(&Cache);
+		  if (BrokenFix == true && DoUpgrade(Pkg) == true)
+		  {
+		     if (Cache[I].InstBroken() == false &&
+			 State.BrokenCount() >= Cache.BrokenCount())
+		     {
+			if (Debug == true)
+			   clog << "  Installing " << Pkg.Name() << endl;
+			Change = true;
+			break;
+		     }
+		     else
+			State.Restore();
+		  }
+
 		  if (Debug == true)
 		     clog << "  Added " << Pkg.Name() << " to the remove list" << endl;
 		  
@@ -1405,7 +1422,7 @@
    return -1;
    
    if (L->Priority != R->Priority)
-      return R->Priority - L->Priority;
+      return L->Priority - R->Priority;
    return strcmp(L.ParentPkg().Name(),R.ParentPkg().Name());
 }
 void pkgPrioSortList(pkgCache &Cache,pkgCache::Version **List)
diff -ur apt-0.5.5cnc4.1/apt-pkg/rpm/rpmpackagedata.h apt-0.5.5cnc4.1_20030324a/apt-pkg/rpm/rpmpackagedata.h
--- apt-0.5.5cnc4.1/apt-pkg/rpm/rpmpackagedata.h	2003-03-08 02:12:32 +0300
+++ apt-0.5.5cnc4.1_20030324a/apt-pkg/rpm/rpmpackagedata.h	2003-03-21 02:30:15 +0300
@@ -67,7 +67,11 @@
    public:
 
    inline pkgCache::State::VerPriority VerPriority(string Package) 
-   	{return Priorities[Package];};
+   {
+      if (Priorities.find(Package) != Priorities.end())
+	 return Priorities[Package];
+      return pkgCache::State::Optional;
+   };
    inline pkgCache::Flag::PkgFlags PkgFlags(string Package) 
    	{return Flags[Package];};
 
